File: persistence/play-jdbc-api/src/main/java/play/db/DBApi.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package play.db;

import java.util.List;

/** DB API for managing application databases. */
public interface DBApi {

  /**
   * @return all configured databases.
   */
  List<Database> getDatabases();

  /**
   * @param name the configuration name of the database
   * @return Get database with given configuration name.
   */
  Database getDatabase(String name);

  /** Shutdown all databases, releasing resources. */
  void shutdown();
}

File: persistence/play-jdbc-api/src/main/scala/play/api/db/DBApi.scala
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package play.api.db

/**
 * DB API for managing application databases.
 */
trait DBApi {

  /**
   * All configured databases.
   */
  def databases(): Seq[Database]

  /**
   * Get database with given configuration name.
   *
   * @param name the configuration name of the database
   */
  def database(name: String): Database

  /**
   * Shutdown all databases, releasing resources.
   */
  def shutdown(): Unit
}
